plugin/storage/cassandra/factory.go

```diff
@@ -50,22 +49,26 @@ var ( // interface comformance checks
 type Factory struct {
 	Options *Options
 
-	primaryMetricsFactory metrics.Factory
-	archiveMetricsFactory metrics.Factory
-	logger                *zap.Logger
-	tracer                trace.TracerProvider
+	metricsFactory metrics.Factory
+	logger         *zap.Logger
+	tracer         trace.TracerProvider
 
-	primaryConfig  config.SessionBuilder
-	primarySession cassandra.Session
-	archiveConfig  config.SessionBuilder
-	archiveSession cassandra.Session
+	config  config.SessionBuilder
+	session cassandra.Session
 }
 
 // NewFactory creates a new Factory.
-func NewFactory() *Factory {
+func NewFactory(isArchive bool) *Factory {
```

Comment:
nit: let's not pass a flag but have a 2nd constructor NewArchiveFactory. Because `NewFactory(true)` has poor readability.